app/docs/components/dropdown/dropdown.mdx

```diff
@@ -123,3 +123,11 @@ The dropdown component can be used to show a list of menu items when clicking on
     </div>
   </div>
 </CodePreview>
+<CodePreview title="Custom trigger dropdown">
+  <Dropdown dismissOnClick={false} CustomTrigger={<button>Custom Dropdown button</button>}>
```

Comment:
I think that this approach would also work. But maybe we can use something like `renderTrigger=` instead? Check the #730

Reply:
@paghar here, what I would prefer, instead passing the `ReactNode` I would prefer to pass a function... using the property `renderTrigger={(theme) => <button></button>}`

The concept is similar to: https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop

Here is a similar implementation: https://github.com/themesberg/flowbite-react/blob/408aaf3ef34b6b8b1c86f79697d1f9e259647ce1/src/components/Card/Card.spec.tsx#L30